### ISSUE TITLE
[Feature]: Support `htmlwidgets`

### ISSUE THREAD

**Issue by dipterix** [coatless/quarto-webr#222]
### Feature Description

Currently there is no support for `htmlwidgets` by `quarto-webr`.

I somehow managed to get it working. Here I would like to propose this method, hopefully it can be integrated into `quarto-webr`.

Basically there are two changed needed to visualize `htmlwidgets` via `webr`:

1. The widget must be able to be saved as `selfcontained`
2. `quarto-webr` needs to support `browse` type (introduced by `webr::viewer_install()`

For Step 1, users can use this [work-around](https://www.jsinr.me/2024/01/10/selfcontained-htmlwidgets/) or wait for `htmlwidgets` to add this feature.
Step 2 needs to be done from `quarto-webr` (see PR https://github.com/coatless/quarto-webr/pull/223).

```r
library(dygraphs)
view <- function(widget) {
	webr::viewer_install()
	path <- save_selfcontained(v)
	getOption("viewer")(path)
}

v <- dygraph(nhtemp, main = "New Haven Temperatures") %>% 
  dyRangeSelector(dateWindow = c("1920-01-01", "1960-01-01"))
view(v)
```

![image](https://github.com/user-attachments/assets/3aa616fc-7d61-4182-a347-9ec79ab8bda4)

**Comment by dipterix** [comment]
Here's an example with the PR:

https://rave.wiki/posts/3dviewer/viewer201.html

**Comment by coatless** [comment]
@dipterix thanks! I'll check out the PR tomorrow in more depth. 

P.S. Woahza, that's a really nice embedding demo.

**Comment by holtzy** [comment]
Would love to see this feature integrated too!

**Comment by coatless** [comment]
Closed by #223 

Note, we probably want to incorporate `save_selfcontained()` proposed by timelyportfolio inside of qwebr but we'll leave that for a second follow-up.

**Comment by dipterix** [comment]
Thanks, this is great!
